enhancements/etcd/cluster-etcd-operator.md

```diff
@@ -178,6 +178,40 @@ items:
           status: Remove
 ```
 
+### Migration of certificates, keys, and CA bundles
+
+In 4.3 (and currently in 4.4), the certificates, keys and CA bundles use to launch etcd are located on disk on the masters.
+To transition to an operator, we need to maintain the source of truth for these files in secrets and configmaps.
+To support migration, we will create a daemonset (`ds/etcd-credential-pusher`) that will create configmaps for the CA bundles and secrets for the configmaps if they are not present.
+The known set of files is currently
+ 1. `etcd --cert-file=/etc/ssl/etcd/system:etcd-server:${ETCD_DNS_NAME}.crt --key-file=/etc/ssl/etcd/system:etcd-server:${ETCD_DNS_NAME}.key`
+    This will be pushed to `oc -n openshift-etcd get secret/etcd-serving-cert`.
+    It doesn't matter which particular pair wins.
+ 2. `etcd --trusted-ca-file=/etc/ssl/etcd/ca.crt`
+    This will be pushed to `oc -n openshift-etcd get configmap/etcd-serving-ca`.
+    It doesn't matter which particular bundle wins.
+ 3. `etcd --peer-cert-file=/etc/ssl/etcd/system:etcd-peer:${ETCD_DNS_NAME}.crt --peer-key-file=/etc/ssl/etcd/system:etcd-peer:${ETCD_DNS_NAME}.key`
+    This will be pushed to `oc -n openshift-etcd get secret/etcd-peer-client-cert`.
+    It doesn't matter which particular pair wins.
```

Comment:
Because we use TLS authentication[1] on the etcd server `--peer-client-cert-auth=true` and `--client-cert-auth=true` the IP and DNS in the SAN for each `Server` and `Peer` listening interface must match that of the caller for auth to succeed.

<client><dial>etcd-1.foo.bar.com  **<--[auth fail]->** <server>:2379<SAN><DNS: etcd-0.foo.bar.com>

# assuming etcd server vs proxy
So if Peer A is using Peer cert in client context against Peer B interface. If the caller is using FQDN `etcd-0.whatever.com:2380` that FQDN must be included in the DNS SAN. The same goes for IP. Or auth will fail.

For this reason we have each etcd generate CSR requests specific to this meta (IP, FQDN)[2]

[1] https://github.com/etcd-io/etcd/blob/master/Documentation/op-guide/security.md#notes-for-tls-authentication
[2] https://github.com/openshift/kubecsr/blob/master/pkg/certagent/agent.go#L93